documentation/dataplane/dash-flow/DASH_FLOW_SAI.md

```diff
@@ -0,0 +1,615 @@
+
+
+# README
+
+DASH Flow SAI abstracts the flow state within DASH. In the quest for achieving cloud services, the control plane can generate a flow state table, performing operations such as adding, deleting, modifying, and querying. This specification guarantees alignment between the flow state table and flow states across different cloud providers and vendors. We propose an API to provide a vendor-neutral method for managing flow states, including uniform control of Programmable switches, SmartNICs, and Smart Switches.
+
+DASH Flow SAI accommodates multiple flow tables and diverse flow entry operations. Primarily, it negates differences between vendors' implementations, offering universal interfaces. However, it acknowledges performance variations between DASH and SAI deployed on smart devices. For example, it supplies APIs to handle single and batch flows, with entry count as a limiting factor.
+
+With DASH Flow SAI, it's possible to achieve varied cloud services requiring flow states. For instance, a cloud load balancer can add a flow decision entry and retrieve the flow entry via the data plane. It also lays the groundwork for DASH to provide basic services, such as high availability.
+
+
+
+# DASH Flow SAI
```

Comment:
Before going to the APIs, we need a section to briefly describe how the flow itself is modeled and what is the relationships between the flow table, flow, reverse flow and ENIs.

For the use case or some behavior of the flows, we can reference the SmartSwitch HA doc the flow sections.

Reply:
With this sections, later, we can organize the rest of doc in several parts (headings as below):

- SAI API design
  - Flow table APIs
  - Flow APIs
    - Basic flow APIs
    - Bulk sync with bulk get APIs
    - Protobuf-based flow programming